### DISCUSSION TITLE
Extract Filename from file upload column into Formula column

### DISCUSSION THREAD

**Discussion by eastendtom** [Budibase/budibase#3078] [Help]
Hi Budibase team,

Excuse the noob question. I'm looking into your rather lovely no-code system with, well, no ability to code myself. But hoping to get a simple framework together for a film editing shot management system which some smart people are building the back-end for.

My first question is this. I've made a simple file uplaoder to a table dedicated to hold data request text files. I want to extract the name of the file upon upload, and name the new record by the filename without the user needing to copy and paste. Is there something in Formula which will let me do this? I can't see a way.

Thanks in advance,
Tom.

**Comment by mjashanks** [comment]
Hi Tom,

Had a play around with this myself. Not possible at the moment.

FYI I tried setting up a binding to `{{ Repeater.Table.AttachmentField.name }}` to try and pull the filename out. Attachments could be improved, but nothing is scheduled to look at them at the moment.

**Reply by eastendtom** [reply]
Hi Michael,

Leaving aside the second question, is there no way to extract the filename? If I hover over the uploaded file it reads the file's name pretty nicely.

Thanks again,
Tom.

<img width="309" alt="Screenshot 2021-10-20 at 12 03 37" src="https://user-images.githubusercontent.com/86465981/138081740-45dc2b9d-a893-4679-978b-0908b329b7c6.png">

**Reply by mjashanks** [reply]
Unfortunately not. All the data is there in the app, there's just no way to display it at the moment - sorry.

**Comment by eastendtom** [comment]
Thank you Michael,

Appreciate your input and experience. That’s a shame. I’ll see if I can fudge something which would help in the short term.

I have some other questions which are a bit more operational than coding. Can I hit you with it?

I have Users which are attached to multiple Projects. Long story short, I’d like to display a drop-down next to their username in the Header so that all viewed tables can be filtered by the active project.

My Users each have a Relationship column linking to a Table of (multiple) Projects, and there’s a User Column as well (Text field) for their Active Project.

I can’t for the life of me setup this behaviour, any bright ideas, please?

Thanks so much,
Tom.
________________________________

**Comment by xnohat** [comment]
Hi , I found a way to extract data from attachment column by using pluck helper

Handlebars:
{{ New Repeater.Table Name.Attachment Column }}
Raw return from attachment column:

[{"size":386095,"name":"image file name with white space.jpg","url":"/prod-budi-app-assets/app_c98fa529298a498d80cd7a9e6213ef06/attachments/12cf20de-962b-4ca8-a7d8-e1a0947528c5.jpg","extension":"jpg","key":"app_c98fa529298a498d80cd7a9e6213ef06/attachments/12cf20de-962b-4ca8-a7d8-e1a0947528c5.jpg"}]

Using pluck helper to extract data:

Extract file name:
{{pluck New Repeater.Table Name.Attachment Column 'name'}}

Extract URL and set Link to download in 'Link URL' of Card Element
https://budibaseappdomain.com{{pluck New Repeater.Table Name.Attachment Column 'url'}}

**Reply by mjashanks** [reply]
`{{New Repeater.Table Name.Attachment Column.0.name}}` should also work
